### ISSUE TITLE
Add pagination support to bank.totalSupply query in BankExtension

### ISSUE THREAD

**Issue by webmaster128** [cosmos/cosmjs#1095]
After #1094 all supported backends have the pagination support in `QueryTotalSupplyRequest` and we should support it directly in `BankExtension` using the pagination key as we do for other list queries.

**Comment by JGJP** [comment]
I would actually recommend that we expand pagination support to other queries also. Specifically for the limit param. Another example that I had to manually override would be Staking/ValidatorDelegations.

Currently extensions that support pagination only accept the pagination key as a param and will form their own Pagination object for the request that sets 0 as the limit, not sure what the intention is here but it's not particularly useful since the RPC node will time out on heavy requests, trying to return all results:
https://github.com/cosmos/cosmjs/blob/e098e763ddc19800ee9408f1160949929af1da1b/packages/stargate/src/queries/utils.ts#L29

I suggest we allow the passing of limit as a regular integer, and then pass that to `createPagination` to use, if provided.

**Comment by JGJP** [comment]
I mean ideally we could pass any of these for it to use, turning any integers to Longs in this utility function. It would not be particularly confusing as long as the arguments are typed exactly as the underlying client Request types are:
https://github.com/confio/cosmjs-types/blob/e3e0c4ba52d38af45522f5705c24eb734494b9a4/src/cosmos/bank/v1beta1/query.ts#L43-L54

We can write a Typescript function that uses the original type as is, except Longs are turned to numbers:
```ts
type QueryTypeAsExtensionParam<QueryType> = {
    [Key in keyof QueryType]: QueryType[Key] extends Long
        ? number
        : QueryType[Key]
}
```

**Comment by webmaster128** [comment]
> object for the request that sets 0 as the limit, not sure what the intention is here but it's not particularly useful since the RPC node will time out on heavy requests, trying to return all results:

In protobuf, the empty/default value 0 cannot be differentiated from unset. So 0 means unset. In the protobuf docs of `interface PageRequest` we have

```ts
  /**
   * limit is the total number of results to be returned in the result page.
   * If left empty it will default to a value to be set by each app.
   */
  limit: Long;
```

So we don't use unlimited by default.

**Comment by webmaster128** [comment]
@JGJP Thanks for all your input. I did not forget your request regarding more detailed pagination settings. But if we do that change, we should do that for all queries that are supported right now. And looking at the evolution of those query wrappers, they will hopefully become obsolete soon. Then you can just use the full protobuf API.

**Comment by webmaster128** [comment]
Shipped as part of CosmJS 0.29.0
